docs/docs/reference-react-component.md

```diff
@@ -0,0 +1,261 @@
+
+#### Mounting
+
+These methods are called when an instance of a component is being created and inserted into the DOM:
+
+- [`constructor(props)`](#constructor)
+- [`componentWillMount()`](#componentwillmount)
+- [`render()`](#render)
+- [`componentDidMount()`](#componentdidmount)
+
+#### Updating
+
+An update can be caused by changes to props or state. These methods are called when a component is being re-rendered:
+
+- [`componentWillReceiveProps(nextProps)`](#componentwillreceivepropsnextprops)
+- [`shouldComponentUpdate(nextProps, nextState)`](#shouldcomponentupdatenextprops-nextstate)
+- [`componentWillUpdate(nextProps, nextState)`](#componentwillupdatenextprops-nextstate)
+- [`render()`](#render)
+- [`componentDidUpdate(prevProps, prevState)`](#componentdidupdateprevprops-prevstate) is invoked after an update and can be used to operate on the DOM after the component has been updated or to fetch new data in response to prop changes.
+
+#### Unmounting
+
+This method is called when a component is being removed from the DOM:
+
+- [`componentWillUnmount()`](#componentwillunmount)
+
+### Other APIs
+
+Each component also provides some other APIs:
+
+  - [`defaultProps`](#defaultprops)
+  - [`displayName`](#displayName)
+  - [`forceUpdate(callback)`](#forceupdatecallback)
+  - [`props`](#props)
+  - [`propTypes`](#proptypes)
+  - [`setState(nextState, callback)`](#setstatenextstate-callback)
+  - [`state`](#state)
+
+## Reference
+
+### `render()`
+
+The `render()` method is required.
+
+When called, it should examine `this.props` and `this.state` and return a single React element. This element can be either a representation of a native DOM component, such as `<div />`, or another composite component that you've defined yourself.
+
+You can also return `null` or `false` to indicate that you don't want anything rendered. When returning `null` or `false`, `ReactDOM.findDOMNode(this)` will return `null`.
+
+The `render()` function should be pure, meaning that it does not modify component state, it returns the same result each time it's invoked, and it does not directly interact with the browser. If you need to interact with the browser, perform your work in `componentDidMount()` or the other lifecycle methods instead. Keeping `render()` pure makes components easier to think about.
+
+> Note
+>
+> `render()` will not be invoked if [`shouldComponentUpdate()`](#shouldcomponentupdatenextprops-nextstate) returns false.
+
+### `constructor(props)`
+
+The constructor for a React component is called before it is mounted. When implementing the constructor for a `React.Component` subclass, you should call `super(props)` before any other statement. Otherwise, `this.props` will be undefined in the constructor, which can lead to bugs.
```

Comment:
Mention early that constructor is not required if you don't use state and don't bind methods?